frameioclient/py3_uploader.py

```diff
@@ -0,0 +1,46 @@
+import requests
+import math
+import concurrent.futures
+import threading
+
+thread_local = threading.local()
+
+class FrameioUploader(object):
+  def __init__(self, asset, file):
+    self.asset = asset
+    self.file = file
+
+  def _read_chunk(self, file, size):
+    while True:
+      data = file.read(size)
+      if not data:
+        break
+      yield data
+
+  def _get_session(self):
+    if not hasattr(thread_local, "session"):
+        thread_local.session = requests.Session()
+    return thread_local.session
+
+  def _upload_chunk(self, task, i, total):
+    print(f"Beginning chunk upload {i}/{total}")
+    url = task[0]
+    chunk = task[1]
+    session = self._get_session()
+
+    session.put(url, data=chunk, headers={
+      'content-type': self.asset['filetype'],
+      'x-amz-acl': 'private'
+    })
+
+    print(f"Completed chunk upload {i}/{total}")
+
+  def upload(self):
+    total_size = self.asset['filesize']
+    upload_urls = self.asset['upload_urls']
+    size = int(math.ceil(total_size / len(upload_urls)))
+
+    with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
```

Comment:
oh this is really nice. 

but how does it work with multiple files?
I see that you're creating a queue of file chunk to be uploaded. Great. But it seems to me that you create a queue for each file. So if someone uploads 100 files, do we have 100 different queues with each 5 concurrent upload streams? Or is the pool/queue a singleton?

Reply:
With multiple files, process forking should keep it working correctly since the upload function is called independently. 

I'm going to setup a quick test though that adds more concurrency and doesn't rely on blocking calls to upload and attempts to parallelize.

Reply:
Just confirmed that this does work as expected, with proper process forking!